Formula/qscintilla2.rb

```diff
@@ -18,89 +17,81 @@ class Qscintilla2 < Formula
     spec << "-arm64" if Hardware::CPU.arm?
     args = %W[-config release -spec #{spec}]
 
-    cd "Qt4Qt5" do
+    xy = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
+
+    cd "src" do
       inreplace "qscintilla.pro" do |s|
+        s.gsub! "QMAKE_POST_LINK += install_name_tool -id @rpath/$(TARGET1) $(TARGET)",
+          "QMAKE_POST_LINK += install_name_tool -id #{lib}/$(TARGET1) $(TARGET)"
         s.gsub! "$$[QT_INSTALL_LIBS]", lib
         s.gsub! "$$[QT_INSTALL_HEADERS]", include
+        # TODO: use qt6 directory layout when octave can migrate to qt6
         s.gsub! "$$[QT_INSTALL_TRANSLATIONS]", prefix/"trans"
         s.gsub! "$$[QT_INSTALL_DATA]", prefix/"data"
         s.gsub! "$$[QT_HOST_DATA]", prefix/"data"
+        # s.gsub! "$$[QT_INSTALL_TRANSLATIONS]", share/"qt/translations"
+        # s.gsub! "$$[QT_INSTALL_DATA]", share/"qt/mkspecs"
+        # s.gsub! "$$[QT_HOST_DATA]", share/"qt"
       end
 
       inreplace "features/qscintilla2.prf" do |s|
         s.gsub! "$$[QT_INSTALL_LIBS]", lib
         s.gsub! "$$[QT_INSTALL_HEADERS]", include
       end
 
-      qt5 = Formula["qt@5"].opt_prefix
-      system "#{qt5}/bin/qmake", "qscintilla.pro", *args
+      qt = Formula["qt@5"].opt_prefix
+      system "#{qt}/bin/qmake", "qscintilla.pro", *args
       system "make"
       system "make", "install"
     end
 
-    # Add qscintilla2 features search path, since it is not installed in Qt keg's mkspecs/features/
-    ENV["QMAKEFEATURES"] = prefix/"data/mkspecs/features"
-
     cd "Python" do
-      (share/"sip").mkpath
-      version = Language::Python.major_minor_version Formula["python@3.9"].opt_bin/"python3"
-      pydir = "#{lib}/python#{version}/site-packages/PyQt5"
-      system Formula["python@3.9"].opt_bin/"python3", "configure.py", "-o", lib, "-n", include,
-                        "--apidir=#{prefix}/qsci",
-                        "--destdir=#{pydir}",
-                        "--stubsdir=#{pydir}",
-                        "--qsci-sipdir=#{share}/sip",
-                        "--qsci-incdir=#{include}",
-                        "--qsci-libdir=#{lib}",
-                        "--pyqt=PyQt5",
-                        "--pyqt-sipdir=#{Formula["pyqt"].opt_share}/sip/Qt5",
-                        "--sip-incdir=#{Formula["sip"].opt_include}",
-                        "--spec=#{spec}",
-                        "--no-dist-info"
-      system "make"
-      system "make", "install"
-      system "make", "clean"
+      mv "pyproject-qt5.toml", "pyproject.toml"
+      Formula["pyqt"].opt_lib
+      (buildpath/"Python/pyproject.toml").append_lines <<~EOS
+        [tool.sip.project]
+        sip-include-dirs = ["#{Formula["pyqt@5"].opt_lib}/python#{xy}/site-packages/PyQt5/bindings"]
+      EOS
+
+      args = %W[
+        --target-dir #{prefix}
+        --no-make
+        --verbose
+
+        --qsci-features-dir #{prefix}/data/mkspecs/features
```

Comment:
These dirs should also be changed when it migrates to Qt6

Reply:
Can you add a comment with the changes that need to be made?